xds/internal/xdsclient/xdsresource/filter_chain.go

```diff
@@ -92,12 +96,17 @@ type RouteWithInterceptors struct {
 	Interceptors []resolver.ServerInterceptor
 }
 
+type RoutingConfiguration struct {
+	VHS []VirtualHostWithInterceptors
+	Err error
+}
+
 // ConstructUsableRouteConfiguration takes Route Configuration and converts it
 // into matchable route configuration, with instantiated HTTP Filters per route.
-func (f *FilterChain) ConstructUsableRouteConfiguration(config RouteConfigUpdate) ([]VirtualHostWithInterceptors, error) {
+func (fc *FilterChain) ConstructUsableRouteConfiguration(config RouteConfigUpdate) ([]VirtualHostWithInterceptors, error) {
```

Comment:
Why does this method not return a value of the above newly defined type. The values that it returns seem to directly correspond to the fields of the above struct.

Reply:
Good point. I take the returned values and wrap it in a RoutingConfiguration at all callsites. Will switch.